mlrun/datastore/datastore.py

```diff
@@ -50,7 +50,7 @@ def parse_url(url):
 
 def schema_to_store(schema):
     # import store classes inside to enable making their dependencies optional (package extras)
-    if not schema or schema in ["file"] + list(string.ascii_lowercase):
+    if not schema or schema in ["file", "kafkaTarget"] + list(string.ascii_lowercase):
```

Comment:
I'm not sure I understand if `kafkaTarget` and `kafkaSource` are supposed to be URL schemes. If so, I don't see any tests that use these URL schemes.

And there's a another issue. URL schemes [are case-insensitive](https://www.rfc-editor.org/rfc/rfc3986#section-3.1).
> Although schemes are case-
   insensitive, the canonical form is lowercase and documents that
   specify schemes must do so with lowercase letters.  An implementation
   should accept uppercase letters as equivalent to lowercase in scheme
   names (e.g., allow "HTTP" as well as "http") for the sake of
   robustness but should only produce lowercase scheme names for
   consistency.

Reply:
This isn't about URLs. When using a DS profile, the schema is stored within the profile itself and not the URL. The sequence of events is as follows:
- Establish a KafkaTarget DS profile named "kfk_target".
- Later, refer to a target using "d://kfk_target".
- The `ingest()` function invokes `get_or_create_store("d://kfk_target")`.
- Within the `get_or_create_store()` function, the target's schema is retrieved and `schema_to_store("kafkaTarget")` is then called.

Reply:
See
```
schema = parsed_url.scheme.lower()
```
in `parse_url()`.
It clearly gets the "schema" out of the URL **scheme**.

I'm not sure how it works when the scheme is shifted to lower case, and then later compared against `"kafkaTarget"` in `schema_to_store()`.

Reply:
And anything on `+ list(string.ascii_lowercase)`? It looks like a (preexisting) bug.

Reply:
> And anything on `+ list(string.ascii_lowercase)`? It looks like a (preexisting) bug.

This is to support windows full path  including drive letter , something like `c:/some/path`

Reply:
Somehow I cannot reply to 'https://github.com/mlrun/mlrun/pull/4088#discussion_r1322396996', so I do it here.
Why kafkaTarget scheme should get us to return a FileStore I don't know - the question goes to the authors of kafka target :). Prior the datastore profile kafka path was not an URL but just `path/to/kafka` that has no schema and that bounced to FileStore.

Reply:
I will change kafkaTarget to kafka_target - maybe in the future we will want to make it a part of URL

Reply:
> This is to support windows full path including drive letter , something like c:/some/path
Can you please add a comment? I thought it might be the case, but it also looks like it might be a bug.